src/ufoLib2/objects/layerSet.py

```diff
@@ -20,7 +21,12 @@
 from ufoLib2.typing import T
 
 
-def _convert_layers(value: Iterable[Layer]) -> "OrderedDict[str, Layer]":
+def _convert_layers(
+    value: Union[Iterable[Layer], "OrderedDict[str, Layer]"]
+) -> "OrderedDict[str, Layer]":
+    # XXX: Remove this converter or convert into private non-converter method? Or do
+    # everything in read()?
```

Comment:
I think we can make an new `from_iterable` classmethod constructor that takes an Iterable[Layer], makes the OrderedDict and constructs a LayerSet with it.
Then we can remove this converter and the default `__init__` constructor will only accept `layers` as OrderedDict (and you can do away with all the `# type: ignore` elsewhere).
And one can still make a LayerSet from a simple list of Layers without needing to import OrderedDict.